actors/miner/src/policy.rs

```diff
@@ -207,13 +211,38 @@ pub fn reward_for_disputed_window_post(
     BASE_REWARD_FOR_DISPUTED_WINDOW_POST.clone()
 }
 
-// The daily fee payable per sector.
-pub fn daily_proof_fee(policy: &Policy, circulating_supply: &TokenAmount) -> TokenAmount {
-    let num = BigInt::from(policy.daily_fee_circulating_supply_multiplier_num);
-    let denom = BigInt::from(policy.daily_fee_circulating_supply_multiplier_denom);
+// Calculate the daily fee for a sector's quality-adjusted power based on the current circulating
+// supply.
+pub fn daily_proof_fee(
+    policy: &Policy,
+    circulating_supply: &TokenAmount,
+    qa_power: &StoragePower,
+) -> TokenAmount {
+    let num = BigInt::from(policy.daily_fee_circulating_supply_qap_multiplier_num);
+    let denom = BigInt::from(policy.daily_fee_circulating_supply_qap_multiplier_scale_1)
+        * BigInt::from(policy.daily_fee_circulating_supply_qap_multiplier_scale_2);
 
-    let fee = (num * circulating_supply.atto()).div_floor(&denom);
-    TokenAmount::from_atto(fee)
+    // num/denom gives us the fraction of the circulating supply that should be paid as a fee per
+    // byte of quality-adjusted power.
+
+    let power_multiplier = (num * &*DAILY_FEE_PRECISION_SCALE * qa_power) / denom;
+    TokenAmount::from_atto(
+        (power_multiplier * circulating_supply.atto()).div_floor(&*DAILY_FEE_PRECISION_SCALE),
+    )
```

Comment:
Isn't this strictly more accurate:?

```suggestion
    TokenAmount::from_atto((num * circulating_supply.atto() * qa_power) / denom);
```